sktime/param_est/lag.py

```diff
@@ -0,0 +1,243 @@
+"""Parameter estimators for autoregressive lag order selection.
+
+This module provides functionality for selecting optimal lag orders in
+autoregressive models using information criteria.
+
+The main class ARLagOrderSelector implements lag order selection
+using various information criteria (AIC, BIC, HQIC) and supports both
+sequential and global search strategies.
+"""
+
+__author__ = ["satvshr"]
+__all__ = ["ARLagOrderSelector"]
+
+from sktime.forecasting.auto_reg import AutoREG
+from sktime.param_est.base import BaseParamFitter
+
+
+class ARLagOrderSelector(BaseParamFitter):
+    """Estimate optimal lag order for autoregressive models using information criteria.
+
+    Implements lag order selection for AR models by comparing
+    different lag specifications using information criteria.
+    Supports both sequential and global search strategies.
+
+    Parameters
+    ----------
+    maxlag : int
+        Maximum number of lags to consider
+    ic : str, default="bic"
+        Information criterion to use for model selection:
+        - "aic" : Akaike Information Criterion
+        - "bic" : Bayesian Information Criterion (default)
+        - "hqic" : Hannan-Quinn Information Criterion
+    glob : bool, default=False
+        If True, searches globally across all lag combinations up to maxlag.
+        If False, searches sequentially by adding one lag at a time.
+    trend : str, default="c"
+        Trend to include in the model:
+        - "n" : No trend
+        - "c" : Constant only
+        - "t" : Time trend only
+        - "ct" : Constant and time trend
+    seasonal : bool, default=False
+        Whether to include seasonal dummies in the model
+    hold_back : int, optional (default=None)
+        Number of initial observations to exclude from the estimation sample
+    period : int, optional (default=None)
+        Period of the data (used only if seasonal=True)
+    missing : str, default="none"
+        How to handle missing values:
+        - "none" : No handling
+        - "drop" : Drop missing observations
+        - "raise" : Raise an error
+
+    Attributes
+    ----------
+    selected_model_ : int or tuple
+        Selected lag order(s) that minimize the information criterion
+    ic_value_ : float
+        Value of the information criterion for the selected model
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.param_est.lag import ARLagOrderSelector
+    >>> y = load_airline()
+    >>> selector = ARLagOrderSelector(maxlag=12, ic="bic")  # doctest: +SKIP
+    >>> selector.fit(y)  # doctest: +SKIP
+    ARLagOrderSelector(...)
+    >>> selector.selected_model_  # doctest: +SKIP
+    3
+    >>> selector.ic_value_  # doctest: +SKIP
+    1234.56
```

Comment:
these numbers need to be the actual values that the code returns

Reply:
Fixed that